src/suites/cts/validation/setScissorRect.spec.ts

```diff
@@ -0,0 +1,90 @@
+export const description = `
+setScissorRect validation tests.
+`;
+
+import { TestGroup } from '../../../framework/index.js';
+
+import { ValidationTest } from './validation_test.js';
+
+// TODO: Move beginRenderPass to a Fixture class.
+export class F extends ValidationTest {
+  textureWidth: number = 16;
+  textureHeight: number = 16;
+
+  beginRenderPass(commandEncoder: GPUCommandEncoder): GPURenderPassEncoder {
+    const attachmentTexture = this.device.createTexture({
+      format: 'rgba8unorm',
+      size: { width: this.textureWidth, height: this.textureHeight, depth: 1 },
+      usage: GPUTextureUsage.OUTPUT_ATTACHMENT,
+    });
+
+    return commandEncoder.beginRenderPass({
+      colorAttachments: [
+        {
+          attachment: attachmentTexture.createView(),
+          loadValue: { r: 1.0, g: 0.0, b: 0.0, a: 1.0 },
+        },
+      ],
+    });
+  }
+}
+
+export const g = new TestGroup(F);
+
+g.test('basic use of setScissorRect', t => {
+  const commandEncoder = t.device.createCommandEncoder();
+  const renderPass = t.beginRenderPass(commandEncoder);
+  renderPass.setScissorRect(0, 0, 1, 1);
+  renderPass.endPass();
+  commandEncoder.finish();
+});
+
+g.test('an empty scissor is not allowed', async t => {
```

Comment:
Haven't reviewed the patch yet but one comment: it would be nice to split up tests like this into totally independent tests. This will allow us to suppress individual failures without losing test coverage of the rest of these codeblocks.

Reply:
Done.